packages/patterns/src/components/rename-pattern-category-modal.js

```diff
@@ -0,0 +1,122 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	Modal,
+	Button,
+	TextControl,
+	__experimentalHStack as HStack,
+	__experimentalVStack as VStack,
+} from '@wordpress/components';
+import { store as coreStore } from '@wordpress/core-data';
+import { useDispatch } from '@wordpress/data';
+import { useState } from '@wordpress/element';
+import { decodeEntities } from '@wordpress/html-entities';
+import { __ } from '@wordpress/i18n';
+import { store as noticesStore } from '@wordpress/notices';
+
+/**
+ * Internal dependencies
+ */
+import { CATEGORY_SLUG } from './category-selector';
+
+export default function RenamePatternCategoryModal( {
+	category,
+	onClose,
+	onError,
+	onSuccess,
+	...props
+} ) {
+	const [ name, setName ] = useState( decodeEntities( category.name ) );
+	const [ isSaving, setIsSaving ] = useState( false );
+
+	const { saveEntityRecord, invalidateResolution } = useDispatch( coreStore );
+
+	const { createErrorNotice, createSuccessNotice } =
+		useDispatch( noticesStore );
+
+	const onRename = async ( event ) => {
+		event.preventDefault();
+
+		if ( ! name || name === category.name || isSaving ) {
+			return;
+		}
+
+		try {
+			setIsSaving( true );
+			setName( '' );
+			onClose?.();
```

Comment:
Any reason why we want to immediately close the modal rather than waiting for the renaming to complete (or error)?